crates/networking/rpc/clients/eth/mod.rs

```diff
@@ -753,42 +701,17 @@ impl EthClient {
             ..Default::default()
         };
 
-        let mut wrapped_tx;
-
         if let Some(overrides_gas_limit) = overrides.gas_limit {
             tx.gas_limit = overrides_gas_limit;
-            Ok(tx)
         } else {
-            let mut retry = 0_u64;
-            while retry < bump_retries {
-                wrapped_tx = WrappedTransaction::EIP1559(tx.clone());
-                match self
-                    .estimate_gas_for_wrapped_tx(&mut wrapped_tx, from)
-                    .await
-                {
-                    Ok(gas_limit) => {
-                        // Estimation succeeded.
-                        tx.gas_limit = gas_limit;
-                        return Ok(tx);
-                    }
-                    Err(e) => {
-                        let error = format!("{e}");
-                        if error.contains("replacement transaction underpriced") {
-                            warn!("Bumping gas while building: already known");
-                            retry += 1;
-                            self.bump_eip1559(&mut tx, 110);
-                            continue;
-                        }
-                        return Err(e);
-                    }
-                }
-            }
-            Err(EthClientError::EstimateGasPriceError(
-                EstimateGasPriceError::Custom(
-                    "Exceeded maximum retries while estimating gas.".to_string(),
-                ),
-            ))
+            let mut wrapped_tx = WrappedTransaction::EIP1559(tx.clone());
+            let gas_limit = self
+                .estimate_gas_for_wrapped_tx(&mut wrapped_tx, from)
+                .await?;
+            tx.gas_limit = gas_limit;
```

Comment:
Couldn't this be done as follows?
```suggestion
            self.set_gas_for_wrapped_tx(&mut wrapped_tx, from).await?;
```